Bottom: 73732076108a4004cc42785a74f87cc9952d2d23
Top:    a130d0899b3d1e671d173ee19a640f368c018b4a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-25 13:08:12 +0000

Refresh of refactor-signature-of-plowspol

---

diff --git a/src/find.c b/src/find.c
index 33af2e2..c2219ee 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3446,7 +3446,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
 
 static int
 drc_callback (DataType *data, LayerType *layer, PolygonType *polygon,
-              int type, void *ptr1, void *ptr2)
+              int type, void *ptr1, void *ptr2, void *userdata)
 {
   char *message;
   Coord x, y;
diff --git a/src/polygon.c b/src/polygon.c
index e63391f..755a625 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1536,8 +1536,8 @@ int
 PlowsPolygon (DataType * Data, int type, void *ptr1, void *ptr2,
               int (*call_back) (DataType *data, LayerType *lay,
                                 PolygonType *poly, int type, void *ptr1,
-                                void *ptr2),
-              void *userddata)
+                                void *ptr2, void *userdata),
+              void *userdata)
 {
   BoxType sb = ((PinType *) ptr2)->BoundingBox;
   int r = 0;
diff --git a/src/polygon.h b/src/polygon.h
index 8d1b2b1..64d9a36 100644
--- a/src/polygon.h
+++ b/src/polygon.h
@@ -63,7 +63,7 @@ int PolygonHoles (PolygonType *ptr, const BoxType *range,
 		  int (*callback) (PLINE *, void *user_data),
                   void *user_data);
 int PlowsPolygon (DataType *, int, void *, void *,
-                  int (*callback) (DataType *, LayerType *, PolygonType *, int, void *, void *),
+                  int (*callback) (DataType *, LayerType *, PolygonType *, int, void *, void *, void *),
                   void *userdata);
 void ComputeNoHoles (PolygonType *poly);
 POLYAREA * ContourToPoly (PLINE *);
